### ISSUE TITLE
Eliminate network hardcoding (related to #1152, #1153, #1154, #1155)

### ISSUE THREAD

**Issue by mromascanu123** [terraform-google-modules/terraform-example-foundation#1226]
### TL;DR

Remove the hardcoding at different levels of IPs and configuration options and instead take the values from a configuration object (which may come from a simple yaml file, or injected by a script). For 3-n-h-a-s an example code and configuration attached. Actually this example attached code (base_env-main.tf.zip) was tested working in a POC on derived TEF fork and removed hardcoding. The actual configuration object coming from the calling module e.g. envs/development (development-main.tf.zip). An example (working & tested) config file attached . Just a working example showing this is possible and even potentially beneficial in terms of consolidating the configuration and streamlining configuration. Just examples (although working) certainly can be optimized or done in other more intelligent way

Attachments:

[base_env-main.tf.zip](https://github.com/terraform-google-modules/terraform-example-foundation/files/15256433/base_env-main.tf.zip)
[development-main.tf.zip](https://github.com/terraform-google-modules/terraform-example-foundation/files/15256434/development-main.tf.zip)
[vpc_config.yaml.zip](https://github.com/terraform-google-modules/terraform-example-foundation/files/15256436/vpc_config.yaml.zip)

### Terraform Resources

```markdown
N/A
```

### Detailed design

```markdown
N/A (see TL;DR*)
```

### Additional information

Related to #1152 #1153 #1154 #1155

**Comment by mromascanu123** [comment]
And now it's closed for good w/o any resolution. Much appreciated how it works.
